File: src/util/SwitchException.java
package util;

/**
 * Runtime exception for switch cases that represent programming error and must never execute. This can defend against
 * code modifications, e.g., adding an enum constant. Use in default case, or case that must never execute.
 */
public class SwitchException extends RuntimeException {

	/**
	 * Constructs new switch exception.
	 *
	 * @param switchValue value of the switch case - the object passed into switch statement.
	 */
	public SwitchException(Object switchValue) {
		super("Illegal switch case (unimplemented or forbidden): " + switchValue);
	}
}

File: src/sp/it/pl/util/SwitchException.kt
package sp.it.pl.util

/**
 * Runtime exception for switch cases, when expressions or if-else branches that represent programming error and must
 * never execute.
 * This can defend against code modifications, e.g., adding an enum constant.
 */
class SwitchException(switchValue: Any): RuntimeException("Illegal switch case on value: $switchValue")